docs/proposals/Ashish_Mahajan_Versioning_Proposal.md

```diff
@@ -0,0 +1,241 @@
+- Apply version-specific configurations
+- Deploy to the correct subfolder in gh-pages
+- Preserve the CNAME file for the custom domain
+
+The workflow will determine the target path based on the branch (main → /, master → /v1.0/) and deploy each version to its dedicated location while maintaining the shared domain.
+
+### 3. Version Selector UI Component
+
+I will implement an accessible, user-friendly version selector component:
+
+1. **Component Location**: Integrated into the Jupyter Book in the left sidebar
+2. **Implementation Strategy**:
+   - DOM-based insertion for compatibility with Jupyter Book
+   - Event-driven interaction handling
+   
+3. **UI Features**:
+   - Current version indicator
+   - Dropdown for all available versions
+   - Version-specific metadata (status)
+   - Visual distinction for latest version
+
+4. **Responsive Design**:
+   - Adapts to all screen sizes
+   - Maintains usability on mobile devices
+   - Follows existing responsive breakpoints
+<br><br>
+![version-selector](togglebutton2.png)
+<br><br>
+### 4. Outdated Version Notification Banner
+
+For non-latest versions (v1.0), I will implement a prominent notification banner:
+
+1. **Banner Design**:
+   - Full-width banner at the top of the page
+   - Distinct background color for visibility (red or any color of your choice)
+   - Clear typography with sufficient contrast
+   - Non-disruptive but immediately noticeable
+
+2. **Banner Content**:
+   - Clear message: "You are viewing an older version (v1.0) of the DISCOVER Cookbook"
+   - Direct link to equivalent page in latest version
+   - Release date information to provide context
+
+3. **Technical Implementation**:
+   - Conditional insertion based on version detection
+   - Responsive design for all screen sizes
+   - Persistent across page navigation within same version
+
+![Banner](banner.png)
+
+This banner ensures users are immediately aware when viewing outdated content while providing a direct path to the latest version.
+
+### 5. Cross-Version Navigation System
+
+I will develop intuitive navigation between versions:
+
+1. **Path Mapping Algorithm**:
+   - Maps equivalent content across versions
+   - Handles renamed or restructured pages
+   - Provides graceful fallbacks when content differs
+
+2. **Implementation Strategy**:
+   - URL path normalization
+   - Content structure analysis
+   - Special case handling for moved content
+
+
+### 6. Integration with Jupyter Book
+
+I will extend the existing Jupyter Book configuration to:
+- Load version management components
+- Integrate with the existing theme
+- Preserve current customizations
+- Support the version selector UI
+
+This will involve updating the _config.yml file to include the necessary JavaScript and CSS files while maintaining compatibility with the existing configuration.
+
+### 7. Implementation Dependencies
+
+- **Step 1** (Version Architecture) is foundational and must be implemented first
```

Comment:
We need Step 0 which is a signed off design. As commented above the version architecture should just be tags.